lua/compe_luasnip/init.lua

```diff
@@ -0,0 +1,64 @@
+local compe = require("compe")
+local Source = {}
+local luasnip_exists, luasnip = pcall(require, "luasnip")
+
+function Source.new()
+  return setmetatable({}, { __index = Source })
+end
+
+function Source.get_metadata(_)
+  return {
+    priority = 10;
+    -- keep Snippets with same trigger but different filetype.
+    dup = true;
+    menu = '[Snippets]';
+  }
+end
+
+function Source.determine(_, context)
+  if not luasnip_exists then
+    error("Cannot add source 'luasnip' without LuaSnip installed!")
+    return {}
+  end
```

Comment:
Please remove this error function.
If the user has a wrong configuration, it will be shown many many times.

Reply:
Ah, nice.